src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRPipelineSpark.java

```diff
@@ -0,0 +1,112 @@
+import org.broadinstitute.hellbender.utils.recalibration.BaseRecalibrationEngine;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationArgumentCollection;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationReport;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.IOException;
+import java.util.List;
+@CommandLineProgramProperties(
+        summary = "This tools performs 2 steps of BQSR - creation of recalibration tables and rewriting of the bam, without writing the tables to disk. ",
+        oneLineSummary = "Both steps of BQSR",
+        usageExample = "BQSRSpark -I in.bam --knownSites in.vcf -O out.bam",
+        programGroup = SparkProgramGroup.class
+)
+/**
+ * BQSR. The final result is analysis-ready reads.
+ */
+public final class BQSRPipelineSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Override
+    public boolean requiresReference() { return true; }
+
+    @Argument(doc = "the known variants", shortName = "knownSites", fullName = "knownSites", optional = true)
+    protected List<String> baseRecalibrationKnownVariants;
+
+    @Argument(doc = "the output bam", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    protected String output;
+
+    @Argument(doc = "If specified, shard the output bam", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
+    private boolean shardedOutput = false;
+
+    @Argument(doc = "the join strategy for reference bases and known variants", shortName = "joinStrategy", fullName = "joinStrategy", optional = true)
+    private JoinStrategy joinStrategy = JoinStrategy.SHUFFLE;
+
+    /**
+     * all the command line arguments for BQSR and its covariates
+     */
+    @ArgumentCollection(doc = "all the command line arguments for BQSR and its covariates")
+    private final RecalibrationArgumentCollection bqsrArgs = new RecalibrationArgumentCollection();
+
+    /**
+     * command-line arguments to fine tune the apply BQSR step.
+     */
+    @ArgumentCollection
+    public ApplyBQSRUniqueArgumentCollection applyBqsrArgs = new ApplyBQSRUniqueArgumentCollection();
+
+    @Override
+    public SerializableFunction<GATKRead, SimpleInterval> getReferenceWindowFunction() {
+        return BaseRecalibrationEngine.BQSR_REFERENCE_WINDOW_FUNCTION;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        final JavaRDD<GATKRead> initialReads = getReads();
+        final VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
+
+        if ( baseRecalibrationKnownVariants.isEmpty() ) { // Warn the user if no dbSNP file or other variant mask was specified
+            throw new UserException.CommandLineException(BaseRecalibrator.NO_DBSNP_EXCEPTION);
+        }
```

Comment:
You forgot to fix this -- `baseRecalibrationKnownVariants` should be marked as `optional = false`, and let the command-line parser do this check for you.